Build issue on macos

Seems to be a build issue for rbsigar.c on macos. Possibly solved by adding ctype.h include to it, but I couldn't figure out how to build the project from source so I have not been able to verify.

```
current directory: /Users/[user]/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/kostya-sigar-2.0.6/bindings/ruby
/Users/[user]/.rbenv/versions/2.6.5/bin/ruby -I /Users/[user]/.rbenv/versions/2.6.5/lib/ruby/2.6.0 -r ./siteconf20201002-5522-2xvgri.rb extconf.rb
Ruby platform=x86_64-darwin19
rbsigar_generated.rx needs update
generating rbsigar_generated.rx
fatal: not a git repository (or any of the parent directories): .git
../../src/sigar_fileinfo.c -> sigar_fileinfo.c
../../src/sigar_util.c -> sigar_util.c
../../src/sigar_signal.c -> sigar_signal.c
../../src/sigar_getline.c -> sigar_getline.c
../../src/sigar_ptql.c -> sigar_ptql.c
../../src/sigar_cache.c -> sigar_cache.c
../../src/sigar.c -> sigar.c
../../src/sigar_format.c -> sigar_format.c
../../src/os/darwin/darwin_sigar.c -> darwin_sigar.c
creating Makefile
current directory: /Users/[user]/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/kostya-sigar-2.0.6/bindings/ruby
make "DESTDIR=" clean
current directory: /Users/[user]/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/kostya-sigar-2.0.6/bindings/ruby
make "DESTDIR="
compiling darwin_sigar.c
darwin_sigar.c:335:23: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
    (*sigar)->ticks = sysconf(_SC_CLK_TCK);
                    ~ ^~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:362:16: warning: returning 'const char [37]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "Failed to open /dev/kmem for reading";
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:364:16: warning: returning 'const char [32]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "/proc filesystem is not mounted";
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:1077:14: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
    num = len/sizeof(*proc);
        ~ ~~~^~~~~~~~~~~~~~
darwin_sigar.c:1467:19: warning: duplicate 'const' declaration specifier [-Wduplicate-decl-specifier]
static const char const thread_states[] = {
                  ^
darwin_sigar.c:1649:11: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
    len = strlen(kargs->ptr);
        ~ ^~~~~~~~~~~~~~~~~~
darwin_sigar.c:1682:31: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
        int alen = strlen(ptr)+1;
            ~~~~   ~~~~~~~~~~~^~
darwin_sigar.c:1717:20: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
        int slen = strlen(ptr);
            ~~~~   ^~~~~~~~~~~
darwin_sigar.c:1848:20: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
        klen = val - ptr;
             ~ ~~~~^~~~~
darwin_sigar.c:1853:16: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
        vlen = strlen(val);
             ~ ^~~~~~~~~~~
darwin_sigar.c:2000:23: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
            int len = strlen(ptr);
                ~~~   ^~~~~~~~~~~
darwin_sigar.c:2055:51: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
                                    (char *)name, strlen(name));
                                                  ^~~~~~~~~~~~
darwin_sigar.c:2217:36: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
    if ((num = sigar_getfsstat(fs, len, MNT_NOWAIT)) < 0) {
               ~~~~~~~~~~~~~~~     ^~~
darwin_sigar.c:2432:13: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
    if (mhz != SIGAR_FIELD_NOTIMPL) {
        ~~~ ^  ~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:2435:17: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
    if (mhz_max != SIGAR_FIELD_NOTIMPL) {
        ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:2438:17: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
    if (mhz_min != SIGAR_FIELD_NOTIMPL) {
        ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:2459:13: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
    if (mhz == SIGAR_FIELD_NOTIMPL) {
        ~~~ ^  ~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:2464:17: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
    if (mhz_max == SIGAR_FIELD_NOTIMPL) {
        ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:2467:17: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
    if (mhz_min == SIGAR_FIELD_NOTIMPL) {
        ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~
darwin_sigar.c:2513:16: warning: comparison of integers of different signs: 'int' and 'unsigned int' [-Wsign-compare]
    for (i=0; i<sigar->ncpu; i++) {
              ~^~~~~~~~~~~~
darwin_sigar.c:2530:31: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
        info->total_sockets = sigar_cpu_socket_count(sigar);
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../include/sigar_util.h:147:18: note: expanded from macro 'sigar_cpu_socket_count'
    (sigar->ncpu < sigar->lcpu) ? sigar->ncpu : \
     ~~~~~~~~~~~ ^ ~~~~~~~~~~~
darwin_sigar.c:2652:27: warning: comparison of integers of different signs: 'int' and 'size_t' (aka 'unsigned long') [-Wsign-compare]
    if (sigar->ifconf_len < len) {
        ~~~~~~~~~~~~~~~~~ ^ ~~~
darwin_sigar.c:2654:29: warning: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int' [-Wshorten-64-to-32]
        sigar->ifconf_len = len;
                          ~ ^~~
darwin_sigar.c:3392:42: warning: incompatible pointer types passing 'uint64_t *' (aka 'unsigned long long *') to parameter of type 'rpc_cnt_t *' (aka 'int *') [-Wincompatible-pointer-types]
    map_nfs_stats((sigar_nfs_v3_t *)nfs, &stats.rpccnt[0]);
                                         ^~~~~~~~~~~~~~~~
darwin_sigar.c:3334:59: note: passing argument to parameter 'rpc' here
static void map_nfs_stats(sigar_nfs_v3_t *nfs, rpc_cnt_t *rpc)
                                                          ^
darwin_sigar.c:3418:42: warning: incompatible pointer types passing 'uint64_t *' (aka 'unsigned long long *') to parameter of type 'rpc_cnt_t *' (aka 'int *') [-Wincompatible-pointer-types]
    map_nfs_stats((sigar_nfs_v3_t *)nfs, &stats.srvrpccnt[0]);
                                         ^~~~~~~~~~~~~~~~~~~
darwin_sigar.c:3334:59: note: passing argument to parameter 'rpc' here
static void map_nfs_stats(sigar_nfs_v3_t *nfs, rpc_cnt_t *rpc)
                                                          ^
darwin_sigar.c:3428:16: warning: returning 'const char [6]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "ether";
               ^~~~~~~
darwin_sigar.c:3430:16: warning: returning 'const char [3]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "tr";
               ^~~~
darwin_sigar.c:3432:16: warning: returning 'const char [5]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "fddi";
               ^~~~~~
darwin_sigar.c:3434:16: warning: returning 'const char [4]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "atm";
               ^~~~~
darwin_sigar.c:3436:16: warning: returning 'const char [5]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "vlan";
               ^~~~~~
darwin_sigar.c:3438:16: warning: returning 'const char [9]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "firewire";
               ^~~~~~~~~~
darwin_sigar.c:3441:16: warning: returning 'const char [7]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "bridge";
               ^~~~~~~~
darwin_sigar.c:3444:16: warning: returning 'const char [8]' from a function with result type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return "unknown";
               ^~~~~~~~~
darwin_sigar.c:3708:9: warning: 'Gestalt' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
    if (Gestalt(gestaltSystemVersion, &version) == noErr) {
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Gestalt.h:123:1: note: 'Gestalt' has been explicitly marked deprecated here
Gestalt(
^
darwin_sigar.c:3708:17: warning: 'gestaltSystemVersion' is deprecated: first deprecated in macOS 10.8 - Use NSProcessInfo's operatingSystemVersion property instead. [-Wdeprecated-declarations]
    if (Gestalt(gestaltSystemVersion, &version) == noErr) {
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Gestalt.h:2075:1: note: '' has been explicitly marked deprecated here
enum {
^
darwin_sigar.c:3710:13: warning: 'Gestalt' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
            Gestalt('sys1' /*gestaltSystemVersionMajor*/, &version_major);
            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Gestalt.h:123:1: note: 'Gestalt' has been explicitly marked deprecated here
Gestalt(
^
darwin_sigar.c:3711:13: warning: 'Gestalt' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
            Gestalt('sys2' /*gestaltSystemVersionMinor*/, &version_minor);
            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Gestalt.h:123:1: note: 'Gestalt' has been explicitly marked deprecated here
Gestalt(
^
darwin_sigar.c:3712:13: warning: 'Gestalt' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
            Gestalt('sys3' /*gestaltSystemVersionBugFix*/, &version_fix);
            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Gestalt.h:123:1: note: 'Gestalt' has been explicitly marked deprecated here
Gestalt(
^
darwin_sigar.c:3739:22: warning: assigning to 'char *' from 'const char [7]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            codename = "Jaguar";
                     ^ ~~~~~~~~
darwin_sigar.c:3742:22: warning: assigning to 'char *' from 'const char [8]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            codename = "Panther";
                     ^ ~~~~~~~~~
darwin_sigar.c:3745:22: warning: assigning to 'char *' from 'const char [6]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            codename = "Tiger";
                     ^ ~~~~~~~
darwin_sigar.c:3748:22: warning: assigning to 'char *' from 'const char [8]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            codename = "Leopard";
                     ^ ~~~~~~~~~
darwin_sigar.c:3751:22: warning: assigning to 'char *' from 'const char [13]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            codename = "Snow Leopard";
                     ^ ~~~~~~~~~~~~~~
darwin_sigar.c:3754:22: warning: assigning to 'char *' from 'const char [5]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            codename = "Lion";
                     ^ ~~~~~~
darwin_sigar.c:3757:22: warning: assigning to 'char *' from 'const char [8]' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            codename = "Unknown";
                     ^ ~~~~~~~~~
45 warnings generated.
compiling rbsigar.c
rbsigar.c:105:13: error: implicitly declaring library function 'isdigit' with type 'int (int)' [-Werror,-Wimplicit-function-declaration]
        if (sigar_isdigit(*pid)) {
            ^
rbsigar.c:96:6: note: expanded from macro 'sigar_isdigit'
    (isdigit(((unsigned char)(c))))
     ^
rbsigar.c:105:13: note: include the header <ctype.h> or explicitly provide a declaration for 'isdigit'
rbsigar.c:96:6: note: expanded from macro 'sigar_isdigit'
    (isdigit(((unsigned char)(c))))
     ^
rbsigar.c:562:40: warning: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Wshorten-64-to-32]
                                  list.number);
                                  ~~~~~^~~~~~
1 warning and 1 error generated.
make: *** [rbsigar.o] Error 1
make failed, exit code 2
```

--- a/bindings/ruby/rbsigar.c
+++ b/bindings/ruby/rbsigar.c
@@ -24,6 +24,7 @@
 #endif
 
 #include <errno.h>
+#include <ctype.h>
 #include "sigar.h"
 #include "sigar_fileinfo.h"
 #include "sigar_format.h"

